src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -0,0 +1,168 @@
+        try {
+            this.signingKey = createJwkFromSettings(settings);
+        } catch (Exception e) {
+            throw ExceptionUtils.createJwkCreationException(e);
+        }
+        this.jwtProducer = jwtProducer;
+        if (settings.get("encryption_key") == null) {
+            throw new IllegalArgumentException("encryption_key cannot be null");
+        } else {
+            this.claimsEncryptionKey = settings.get("encryption_key");
+        }
+        if (timeProvider.isPresent()) {
+            this.timeProvider = timeProvider.get();
+        } else {
+            this.timeProvider = () -> System.currentTimeMillis() / 1000;
+        }
+        // CS-SUPPRESS-SINGLE: RegexpSingleline get Extensions Settings
+        this.bwcModeEnabled = settings.getAsBoolean(
+            ConfigConstants.EXTENSIONS_BWC_PLUGIN_MODE,
+            ConfigConstants.EXTENSIONS_BWC_PLUGIN_MODE_DEFAULT
+        );
+        // CS-ENFORCE-SINGLE
+    }
+
+    /*
+     * The default configuration of this web key should be:
+     *   KeyType: OCTET
+     *   PublicKeyUse: SIGN
+     *   Encryption Algorithm: HS512
+     * */
+    static JsonWebKey createJwkFromSettings(Settings settings) throws Exception {
+        String signingKey = settings.get("signing_key");
+
+        if (!Strings.isNullOrEmpty(signingKey)) {
+
+            JsonWebKey jwk = new JsonWebKey();
+
+            jwk.setKeyType(KeyType.OCTET);
+            jwk.setAlgorithm("HS512");
+            jwk.setPublicKeyUse(PublicKeyUse.SIGN);
+            jwk.setProperty("k", signingKey);
+
+            return jwk;
+        } else {
+            Settings jwkSettings = settings.getAsSettings("jwt").getAsSettings("key");
+
+            if (jwkSettings.isEmpty()) {
+                throw new Exception("Settings for key is missing. Please specify at least the option signing_key with a shared secret.");
+            }
+
+            JsonWebKey jwk = new JsonWebKey();
+
+            for (String key : jwkSettings.keySet()) {
+                jwk.setProperty(key, jwkSettings.get(key));
+            }
+
+            return jwk;
+        }
+    }
+
+    public String createJwt(
+        String issuer,
+        String subject,
+        String audience,
+        Integer expirySeconds,
+        List<String> roles,
+        List<String> backendRoles
+    ) throws Exception {
+        long timeMillis = timeProvider.getAsLong();
+        Instant now = Instant.ofEpochMilli(timeProvider.getAsLong());
+
+        jwtProducer.setSignatureProvider(JwsUtils.getSignatureProvider(signingKey));
+        JwtClaims jwtClaims = new JwtClaims();
+        JwtToken jwt = new JwtToken(jwtClaims);
+
+        jwtClaims.setIssuer(issuer);
+
+        jwtClaims.setIssuedAt(timeMillis);
+
+        jwtClaims.setSubject(subject);
+
+        jwtClaims.setAudience(audience);
+
+        jwtClaims.setNotBefore(timeMillis);
+
+        expirySeconds = (expirySeconds == null) ? 300 : expirySeconds;
```

Comment:
300 should be defined somewhere, default max expiration?  Also it doesn't look like we are enforcing an upper limit, which should likely be 300.

Reply:
Yes, I have added them.